onnxruntime/core/optimizer/constant_folding.cc

```diff
@@ -36,9 +47,19 @@ Status ConstantFolding::ApplyImpl(Graph& graph, bool& modified, int graph_level,
       continue;
     }
 
+    // override the EP while setting up OptimizerExecutionFrame::Info so that it will use the CPU kernel for Compute.
+    if (!cpu_ep) {
+      node->SetExecutionProviderType(kCpuExecutionProvider);
```

Comment:
Should this action and the toggle back to the original EP on line 60 be logged ?

Reply:
I don't think so. It's a very temporary thing that is undone. Not sure what a user would do with the information that it happened. If constructing the Info fails we bail out completely as it throws and the initialization fails (returning an error message saying why constructing Info failed).